app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/integration/NgVpnNetworkStack.kt

```diff
@@ -210,18 +212,27 @@ class NgVpnNetworkStack @Inject constructor(
             }
 
             var thread = tunnelThread
+            var remainingAttempts = 5
             while (thread != null && thread.isAlive) {
+                if (remainingAttempts <= 0) {
+                    logcat { "Could not stop tunnel thread" }
+                    deviceShieldPixels.reportTunnelThreadStopTimeout()
+                    break
+                }
```

Comment:
_(See main comment)_ I feel strongly about not touching this file. ie.

This change doesn't really do anything other than `break`ing and leaving the thread hanging. So not really fixing anything and eventually things will break but in a different place much harder to detect.